src/v/cloud_topics/reconciler/tests/range_batch_consumer_test.cc

```diff
@@ -43,11 +58,27 @@ TEST(RangeBatchConsumer, MultipleBatchesRecords) {
     for (int i = 1; i < 4; i++) {
         for (int j = 1; j < 2; j++) {
             auto reader = make_reader(base_offset, record_size, i, j);
-            auto range
-              = std::move(reader).consume(consumer{}, model::no_timeout).get();
-            ASSERT_TRUE(range.has_value());
-            ASSERT_EQ(range->info.base_offset(), base_offset);
-            ASSERT_EQ(range->info.last_offset(), base_offset + (i * j) - 1);
+            iobuf output;
+            auto builder = object_builder::create(
+              make_iobuf_ref_output_stream(output), object_builder::options{});
```

Comment:
It is probably useful to have a test that is close to it's intended usecase: a single object builder, passed to multiple consumers, then verify at the end the incrementally collected state is similar to what the builder recorded.